### PR TITLE
migrate to patternfly/react-topology

### PR DESCRIPTION
**Pull request by christianvogt**
**Fixes**: 
https://issues.redhat.com/browse/ODC-4102

**Analysis / Root cause**: 
Topology package has been migrated to `patternfly/react-topology`.

**Solution Description**: 
Remove topology package from console.
Update all imports.
Update to align with API changes.
 - Use `VisualizationProvider` to set controller.
 - Update `useSelection` which now accepts options object instead of multiple params.
 - Update `EventingPubSubLink.tsx` to fix drawing of the trigger filter decoration because `getPathAlongTheLine` is not in pf-topology. Chose to not move code to a new util because there was issues with how the decorator was rendered.

**Screen shots / Gifs for design review**: 

Before:
image
After:
image

**Browser conformance**: 

- [ ] Chrome
- [ ] Firefox
- [x] Safari
- [x] Edge

/assign @jeff-phillips-18 

_Requires updating to latest patternfly_

### CODE REVIEW

File: frontend/packages/dev-console/src/components/topology/components/nodes/WorkloadNode.tsx [hunk 1]
```diff
@@ -73,7 +73,9 @@ const ObservedWorkloadNode: React.FC<WorkloadNodeProps> = ({
         content={tipContent}
         trigger="manual"
         isVisible={dropTarget && canDrop}
-        tippyProps={{ duration: 0, delay: 0 }}
+        animationDuration={0}
+        entryDelay={0}
+        exitDelay={0}
```
Comment (dtaylor113) [inline frontend/packages/dev-console/src/components/topology/components/nodes/WorkloadNode.tsx:78]: entryDelay & exitDelay default to 0, so you can remove these.

**Comment by christianvogt** [comment]
waiting on https://github.com/openshift/console/pull/5987

**Pull request merged**